### ISSUE TITLE
Tests frequently raise TimedOutException on Travis

### ISSUE THREAD

**Issue by pelson** [SciTools/iris#2618]
As it says on the tin. An example: https://travis-ci.org/SciTools/iris/jobs/246737137

Anybody know when this started happening? Have we taken a performance hit somewhere?

**Comment by pelson** [comment]
Does the following relate?

```
HDF5-DIAG: Error detected in HDF5 (1.8.18) thread 139916343748352:
  #000: H5F.c line 522 in H5Fcreate(): unable to create file
    major: File accessibilty
    minor: Unable to open file
  #001: H5Fint.c line 1003 in H5F_open(): unable to open file: time = Sun Jun 25 09:05:04 2017
, name = '/tmp/tmpaN3ua0/tmp.nc', tent_flags = 13
    major: File accessibilty
    minor: Unable to open file
  #002: H5FD.c line 993 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #003: H5FDsec2.c line 339 in H5FD_sec2_open(): unable to open file: name = '/tmp/tmpaN3ua0/tmp.nc', errno = 13, error message = 'Permission denied', flags = 13, o_flags = 242
    major: File accessibilty
    minor: Unable to open file
```

**Comment by pp-mo** [comment]
I think this is "just" current slowness onTravis

Which is not to say we wouldn't be better off with more streamlined testing...

I spent a large chunk of yesterday afternoon respinning due to timeouts, then things came unstuck about 5pm + seemed to go okay after that.

I tried quite a few things yesterday.
It doesn't seem to be anything in particular about recent commits to Iris
On desktop, things take much the same time as a while back
 \-- even to before mergeback of dask

**Comment by lbdreyer** [comment]
For reference #2633 is attempting to address this

**Comment by pp-mo** [comment]
> For reference #2633 is attempting to address this

It's not really looking terribly promising though.
[summary](https://github.com/SciTools/iris/pull/2633#issuecomment-313424049)

**Comment by pelson** [comment]
> For reference #2633 is attempting to address this

This is about a single test case running slower than the limit (600s), not about how long it takes to setup the Travis container to actually run the tests, so I don't think this *is* being looked at.

**Comment by pelson** [comment]
Just did a little test to compare the difference, and don't get the same slowdown on travis-ci.org/pelson/iris for the py3 default branch:

* SciTools (with a build matrix): https://travis-ci.org/SciTools/iris/jobs/265662089
* Mine (without a build matrix): https://travis-ci.org/pelson/iris/builds/266855557

SciTools: 13 mins with timeouts
Mine: 9 mins without timeouts

Going to re-run mine with all of the matrix items.

**Comment by pelson** [comment]
Like for like commits on master, total run time:
 * pelson: 53 mins https://travis-ci.org/pelson/iris/builds/266863861
 * SciTools: 58 mins https://travis-ci.org/SciTools/iris/builds/256818951

Mostly small diffs - py2.7 default testing takes about a minute longer on SciTools (8.5 mins vs 9.5mins). Almost the exact opposite is observed for py3.5 (so it's quicker on SciTools). Clearly there is a lot of variation!

**Comment by pelson** [comment]
So clearly this is a bigger problem on the Dask branch than it is on master. With that in mind, I pushed that to my Travis instance and got 13 timeout errors. I removed the default dask scheduler line in https://github.com/SciTools/iris/blob/master/lib/iris/_lazy_data.py#L51 and got 1. 

Before: https://travis-ci.org/pelson/iris/builds/266878170
After: https://travis-ci.org/pelson/iris/builds/267073577

**Comment by pelson** [comment]
Closed by #2757.
